File: app/src/main/java/com/shuneault/netrunnerdeckbuilder/game/Cycle.java
package com.shuneault.netrunnerdeckbuilder.game;

import org.json.JSONObject;

public class Cycle {
    public static final String KEY_CODE = "code";
    public static final String KEY_ROTATED = "rotated";

    private String code;
    private boolean rotated;

    public Cycle(JSONObject json) {
        this.code = json.optString(KEY_CODE, "");
        this.rotated = json.optBoolean(KEY_ROTATED, false);
    }

    public Cycle() {

    }

    public String getCode() {
        return code;
    }

    public boolean isRotated() {
        return rotated;
    }

    public void setCode(String code) {
        this.code = code;
    }

    public void setRotation(boolean value) {
        this.rotated = value;
    }
}

File: app/src/main/java/com/shuneault/netrunnerdeckbuilder/game/Cycle.kt
package com.shuneault.netrunnerdeckbuilder.game

import org.json.JSONObject

class Cycle {
    var code: String? = null
    var isRotated = false
        private set

    constructor(json: JSONObject) {
        this.code = json.optString(KEY_CODE, "")
        isRotated = json.optBoolean(KEY_ROTATED, false)
    }

    constructor() {}

    fun setRotation(value: Boolean) {
        isRotated = value
    }

    companion object {
        const val KEY_CODE = "code"
        const val KEY_ROTATED = "rotated"
    }
}